master/pkg/model/task_container_defaults.go

```diff
@@ -20,20 +19,28 @@ import (
 
 // TaskContainerDefaultsConfig configures docker defaults for all containers.
 type TaskContainerDefaultsConfig struct {
+	// These can only be configured in task_container_defaults.
 	DtrainNetworkInterface string                `json:"dtrain_network_interface,omitempty"`
 	NCCLPortRange          string                `json:"nccl_port_range,omitempty"`
 	GLOOPortRange          string                `json:"gloo_port_range,omitempty"`
-	ShmSizeBytes           int64                 `json:"shm_size_bytes,omitempty"`
 	NetworkMode            container.NetworkMode `json:"network_mode,omitempty"`
-	CPUPodSpec             *k8sV1.Pod            `json:"cpu_pod_spec"`
-	GPUPodSpec             *k8sV1.Pod            `json:"gpu_pod_spec"`
-	Image                  *RuntimeItem          `json:"image,omitempty"`
-	RegistryAuth           *types.AuthConfig     `json:"registry_auth,omitempty"`
-	ForcePullImage         bool                  `json:"force_pull_image,omitempty"`
-
-	AddCapabilities  []string      `json:"add_capabilities"`
-	DropCapabilities []string      `json:"drop_capabilities"`
-	Devices          DevicesConfig `json:"devices"`
+
+	// These can be overridden by the environment.pod_spec field in the tasks.
+	CPUPodSpec *k8sV1.Pod `json:"cpu_pod_spec"`
+	GPUPodSpec *k8sV1.Pod `json:"gpu_pod_spec"`
+
+	// These can be overridden by the environment.* field in the tasks.
+	Image            *RuntimeItem      `json:"image,omitempty"`
+	RegistryAuth     *types.AuthConfig `json:"registry_auth,omitempty"`
+	ForcePullImage   bool              `json:"force_pull_image,omitempty"`
+	AddCapabilities  []string          `json:"add_capabilities"`
+	DropCapabilities []string          `json:"drop_capabilities"`
+
+	// This can be overridden by the resources.shm_size field configured in the tasks.
+	ShmSizeBytes int64 `json:"shm_size_bytes,omitempty"`
```

Comment:
It looks like in the final form of this PR, `ShmSizeBytes` is literally never used.  It should be merged into the config object when we merge other fields into the experiment config, and then we should always read it form there.

Reply:
Nope!  I messed up my `git grep`'ing.  This is fine.